File: app/src/androidTest/java/net/tlalka/android/fiszki/test/ActivityMatchers.java
package net.tlalka.android.fiszki.test;

import android.support.test.espresso.matcher.BoundedMatcher;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import org.hamcrest.Description;
import org.hamcrest.Matcher;

public class ActivityMatchers {

    public static Matcher<View> textMatches(String pattern) {
        return new BoundedMatcher<View, TextView>(TextView.class) {

            @Override
            public void describeTo(Description description) {
                description.appendText("text pattern matches");
            }

            @Override
            public boolean matchesSafely(TextView textView) {
                return textView.getText().toString().matches(pattern);
            }
        };
    }

    public static Matcher<View> isDrawable() {
        return new BoundedMatcher<View, ImageView>(ImageView.class) {

            @Override
            public void describeTo(Description description) {
                description.appendText("has drawable image");
            }

            @Override
            public boolean matchesSafely(ImageView imageView) {
                return imageView.getDrawable() != null;
            }
        };
    }
}

File: app/src/androidTest/java/net/tlalka/android/fiszki/test/ActivityMatchers.kt
package net.tlalka.android.fiszki.test

import android.support.test.espresso.matcher.BoundedMatcher
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import org.hamcrest.Description
import org.hamcrest.Matcher

object ActivityMatchers {

    fun isDrawable(): Matcher<View> {
        return object : BoundedMatcher<View, ImageView>(ImageView::class.java) {

            override fun describeTo(description: Description) {
                description.appendText("has drawable image")
            }

            public override fun matchesSafely(imageView: ImageView): Boolean {
                return imageView.drawable != null
            }
        }
    }

    fun textMatches(pattern: String): Matcher<View> {
        return object : BoundedMatcher<View, TextView>(TextView::class.java) {

            override fun describeTo(description: Description) {
                description.appendText("text pattern matches")
            }

            public override fun matchesSafely(textView: TextView): Boolean {
                return textView.text.toString().matches(pattern.toRegex())
            }
        }
    }
}
